resources/js/Components/LanguageSelector.vue

```diff
@@ -1,140 +1,143 @@
-			);
-		},
-	},
-	methods: {
-		onInput( searchedLanguage: string ): void {
-			this.searchInput = searchedLanguage;
-			this.highlightedIndex = 0;
-		},
-		onSelect( languageCode: string ): void {
-			this.$emit( 'select', languageCode );
-		},
-		onClearInputValue(): void {
-			this.searchInput = '';
-		},
-		onCloseMenu(): void {
-			this.$emit( 'close' );
-		},
-		// eslint-disable-next-line vue/no-unused-properties -- exported method
-		focus(): void {
-			const inputRef = this.$refs.input as InstanceType<typeof HTMLInputElement>
-			inputRef.focus();
-		},
-		onArrowDown(): void {
-			this.highlightedIndex = ( this.highlightedIndex + 1 ) % this.shownLanguages.length;
-		},
-		onArrowUp(): void {
-			const length = this.shownLanguages.length;
-			this.highlightedIndex = ( this.highlightedIndex + length - 1 ) % length;
-		},
-		onEnter(): void {
-			this.onSelect( this.shownLanguages[ this.highlightedIndex ].code );
-		},
-	},
-} );
-</script>
+import closeUrlSvg from '../../img/close.svg';
+
+import { ref, computed } from "vue";
+import type { Ref } from 'vue';
+import languageData from "@wikimedia/language-data";
+
+const searchInput: Ref<string> = ref('');
+const highlightedIndex: Ref<number> = ref(-1);
+const closeUrl = ref(closeUrlSvg);
+
+const input = ref<InstanceType<typeof LanguageSelectorInput> | null>(null);
+
+const emit = defineEmits(['select', 'close']);
+
+const languages = computed<Language[]>(() => {
+  const autonyms = languageData.getAutonyms();
+  const languageCodes = Object.keys( autonyms );
+  languageCodes.sort( languageData.sortByAutonym );
+  return languageCodes.map( ( code ) => ( {
+    code,
+    autonym: autonyms[ code ],
+  } ) );
+});
+
+const shownLanguages = computed<Language[]>(() => {
+  return languages.value.filter( ( language ) =>
+      language.code.startsWith( searchInput.value.toLowerCase() ) ||
+      language.autonym.toLowerCase().includes( searchInput.value.toLowerCase() ),
+  );
+});
+
+function onInput(searchedLanguage: string): void {
+  searchInput.value = searchedLanguage;
+  highlightedIndex.value = 0;
+}
+
+function onSelect(languageCode: string): void {
+  emit('select', languageCode);
+}
+
+function onClearInputValue(): void {
+  searchInput.value = '';
+}
+
+function onCloseMenu(): void {
+  emit('close');
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function focus(): void {
+  input.value?.focus();
+}
+
+function onArrowDown(): void {
+  highlightedIndex.value = ( highlightedIndex.value + 1 ) % shownLanguages.value.length;
+}
+
+function onArrowUp(): void {
+  const length = shownLanguages.value.length;
+  highlightedIndex.value = ( highlightedIndex.value + length - 1 ) % length;
+}
 
+function onEnter(): void {
+  onSelect( shownLanguages.value[ highlightedIndex.value ].code )
+}
+
+defineExpose({focus})
+
+</script>
 <style lang="scss">
 $tinyViewportWidth: 38em;
 
 .mismatchfinder__language-selector {
-	position: absolute;
-	inset-inline-end: 0;
```

Comment:
why is the tab size different now?

Reply:
I don't know, we don't have any linting rules regarding these kind of things :(

Reply:
> I don't know, we don't have any linting rules regarding these kind of things :(

but if we don't have linting rules, what caused it to change? are you using prettier on your local IDE? I would suggest to try to get rid of the new tabbing for now in the PR because this is probably irrelevant for the task. maybe we should address that in a different task if it's needed :pray:

Reply:
It's probably because of my IDE's default settings, it wasn't my choice anyways. And you are right, I will get rid of them

Reply:
thanks :)

Reply:
now, it's done :)